### DISCUSSION TITLE
Custom Regex rule based on file contents

### DISCUSSION THREAD

**Discussion by greggathy** [FlorianWoelki/obsidian-icon-folder#121] [Ideas]
Love this plugin! I've got some custom icon rules based on the file and folder names working well. It would be cool to also allow icons to be added based on a regex match of the file contents. 

E.g. when I save webpage content on a new file, the first 5 lines are always a frontmatter block like this:
```
---
page-title: "Page title goes here"
url: helloworld.com
date: "2022-10-19 13:43:40"
---
```

A regex rule could match files that start with that block and I could have a 'website' icon in the custom rule. It would be a great quick-glance view across my entire vault of which files came from webpages

**Comment by FlorianWoelki** [comment]
Great idea yeah. Currently, it's not possible to match a rule against the file contents because it requires to read all the content of all files, which can be really heavy to compute.
